### DISCUSSION TITLE
Navigation Bar

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#267] [Component suggestions]
<img width="480" alt="visuele weergave van het navigation bar component" src="https://github.com/nl-design-system/backlog/assets/84851545/0cad2bf8-86af-4e26-b96f-04f68eff9c8b">

We willen graag meer leren over het Navigation Bar component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1718615560063469)

---

## Naam
Navigation Bar

## Doel
Toont een hiërarchische horizontale navigatie aan de bovenkant van een pagina en biedt de mogelijkheid om tussen pagina’s te navigeren. 

## Ook bekend als
Navbar

## Gerelateerde componenten
[Side Navigation](https://github.com/orgs/nl-design-system/discussions/242)

## Onderzoek
[Menu-Design Checklist: 17 UX Guidelines - NN Group](https://www.nngroup.com/articles/menu-design/)
Artikel waarin best practices worden gedeeld over het ontwerpen van een menu. 

[‘Main Navigation’ Design Examples - Baymard Institute](https://baymard.com/ecommerce-design-examples/27-drop-down-menu)
Onderzoek toont aan dat op hover gebaseerde menu's (ook wel 'megamenu's' genoemd) worden momenteel gebruikt als hoofdnavigatie op 88% van alle Amerikaanse e-commercesites. Toch blijkt uit hun onderzoeken dat megamenu's de grootst mogelijke aandacht vereisen voor interactiedetails als ze moeten goed presteren voor gebruikers.

[Navigation: You Are Here - NN Group](https://www.nngroup.com/articles/navigation-you-are-here/)
Artikel dat dieper ingaat op het weergeven van een visuele indicator voor de huidige pagina.

[How Many Items in a Navigation Menu? - NN Group](https://www.nngroup.com/videos/number-items-navigation-menu/)
Video waarin wordt uitgelegd dat er geen maximum van 7 menu-items nodig is, ondanks een veelvoorkomende mythe.

**Comment by Rozerinay** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-navigation-page-menu--docs)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/page-menu)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=4840-1670&t=GNhHgueBYCiOI86E-11)

---

### Screenshots / Varianten

**Storybook**

Default
Standaard menu items die aan de linkerkant worden uitgelijnd.
<img width="374" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/72caacd0-3bf7-41a8-8b23-c5d266a89b06">

Alignment
Menu items die ten opzichte van de Default variant aan de rechterkant worden uitgelijnd.
<img width="680" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/8c12358d-1b93-4482-b4a8-9dcd6bd0556f">

Wrapping
Maakt het mogelijk om de menu items over meerdere regels te laten lopen. 
<img width="645" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/c2efdb27-3de7-4557-adbb-5f759ef79b70">


**Figma**

States
<img width="1021" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/f69ef74c-3551-4aac-b593-bcf375a19a79">

**Website**
Amsterdam.nl
<img width="341" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/8cbce507-f91b-4886-a83a-c2151e11753f">

Onderzoek.amsterdam.nl
<img width="306" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/039dc0c6-6814-48a4-b8d1-c07821178fa1">


---

### Notities
- Onder de naam Page Menu (en zal Navigation bar heten).
- Het page menu item bestaat uit het Page menu link component.
- Is een unordered list.
- Uitwerking in Storybook en Figma verschilt van de website, omdat het design system nog niet wordt gebruikt voor de website. 
- Menu items kunnen over meerdere regels worden verspreid.  
- Documentatie beschikbaar.

**Reply by VincentSmedinga** [reply]
Hoi Rozerin, de huidige amsterdam.nl gebruikt het design system nog helemaal niet. Er is een volledig nieuwe website onderweg die dat wel doet.

Goed om te weten dat dit component Navigation Bar heet – we waren al op zoek naar iets beters dan Page Menu. (Aanvankelijk hadden we ook Page Grid, Page Header, Page Footer en we dachten alles op dit niveau te groeperen.) We nemen deze naam over.

Het Page Menu Link component dat je noemt is een lokaal subcomponent, dus niet hetzelfde als de generieke Link.

**Reply by Rozerinay** [reply]
@VincentSmedinga heel erg bedankt voor je input! Ik heb de notities aangepast op basis van je opmerkingen. Goed om dit allemaal te weten.

**Comment by Rozerinay** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/react-menu--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Menu)
- [Figma](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/🌻-Den-Haag-Design-System?node-id=1923-4663&t=d5oWbGm5RHvbx0me-11)

---

### Screenshots / Varianten

**Storybook**

Default
Standaard state waarbij je de menu button kunt gebruiken om te navigeren naar een pagina of een menu te openen. 
<img width="230" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/c8307cd2-4abc-4dd8-b38d-65198093c820">

Active
De state wanneer de gebruiker zich op die pagina bevindt. 
<img width="215" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/f8f2f009-ec95-4a5d-9f8f-188984932e8c">


**Figma - Huidig**

Desktop
Bij desktop view zijn de volgende varianten aanwezig: default, icon right, icon left, icon only. 
<img width="911" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/8b9c0798-91d4-41c5-8293-c3ffe2b81e0f">

Mobile
Bij mobile view zijn er geen varianten. 
<img width="1021" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/05949dc8-07de-4072-aee4-fe691bce664e">

**Website**
<img width="489" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/e647fc6c-8f24-484a-b103-06b46c07539e">


---

### Notities
- Onder de naam Menu in Storybook en Menu buttons in Figma.
- Er wordt gebruik gemaakt van een chevron down om het panel te openen.
- Menu items hebben hebben bij default state een donkergrijze kleur.
- Active state gaat uit van de pagina waarop de gebruiker zich bevindt.
- Documentatie beschikbaar.

**Comment by Rozerinay** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/62.8.4/components/navigatie/navbar#sub-navigatie)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/navbar)
- [Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-navbar--primary)

---

### Screenshots / Varianten

**Storybook**

Primary
Hoofdnavigatie (of primaire navigatie): wordt gebruikt wanneer je een website hebt met meerdere applicaties, thema's of onderwerpen hebt zodat gebruikers makkelijk kunnen navigeren tussen pagina's.
<img width="851" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/090356f4-3a66-4a6f-a053-d2ea3827711a">

Secondary
Sub-navigatie: heeft een pagina/onderwerp subcategorieën en is het essentieel om hierbinnen te kunnen navigeren.
<img width="345" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/ce4e636d-a616-4392-9ef3-d88cc4161791">

With extension
Bevat een zoekbalk die onderdeel uitmaakt van de navigatie.
<img width="778" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/f309e18e-7e43-41f3-b149-1c5f7c8c06bb">

**Website**
<img width="1150" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/35ba2ef7-232b-4def-93e5-8c5996e6f138">

Mogelijkheid tot dropdown menu
<img width="626" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/71f21bae-45a0-4865-80b2-98ca85b820ad">

---

### Notities
- Onder de naam Navbar.
- Bestaat uit een hoofd- en subnavigatie.
- Kan bestaan uit menu item 'Meer' waarna je een dropdown menu kunt openen.
- Er wordt gebruik gemaakt van een chevron down om het dropdown menu te openen.
- Is een unordered list waarvan menu items bestaan uit list items. 
- Op small containers verdwijnt de Navigation bar achter een menu button. 
- Documentatie beschikbaar.

**Comment by Rozerinay** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/navigation)

---

### Screenshots / Varianten

**Documentatie website**

Simpele navigatiebalk
Een simpele navigatiebalk gebruik je wanneer er weinig pagina zijn om naar toe te navigeren. Er is een mogelijkheid om een submenu toe te voegen.
<img width="850" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/755d1874-572a-48ab-874a-af4448b7c0a4">

Navigatiebalk met breadcrumbs en menu
Op mobiel is bij de breadcrumbs alleen het laatste niveau en de home-icon zichtbaar. De andere niveaus zijn voor blinde mensen wel zichtbaar. Het hamburgermenu is een collapsable, dus je moet wel de collapse javascript toevoegen (zie de pagina van de collapse).
<img width="835" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/17cce746-9024-41c1-a776-cff5f8d23659">

Navigatiebalk met breadcrumbs en knoppen
Navigatie waarbij de Breadcrumb navigation en knoppen aanwezig zijn, zoals de zoekbutton.
<img width="836" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/4c61e7f4-72a9-4678-949b-870c2fe193a1">

**Website**
Small container
<img width="587" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/05de0f9a-2ca7-4143-b1e8-061920f5fe48">

Collapse details
<img width="1217" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/9d3c037f-5b78-4a85-90b2-951352086e84">

---

### Notities
- Onder de naam Navigation. 
- Bestaat naast menu items uit een breadcrumb navigation en knoppen.
- Er is een mogelijkheid om een submenu toe te voegen.
- Op small containers zijn buttons te zien waarin een icoon en tekst staan van het menu item. De icoon en het label worden onder elkaar gepositioneerd.
- Kan bestaan uit een menu item dat kan worden opengeklapt waarna een collapse details verschijnt.
- Wanneer collapse details aanwezig is, is deze op small containers samen met de knoppen achter een hamburgermenu te vinden.
- Documentatie beschikbaar.

**Comment by Rozerinay** [comment]
## OpenGemeenten

[Patternlab](https://patternlab.lenteveld.nl/?p=molecules-OpenGemeentenSite-PageHeader-Menu)

---

### Screenshots / Varianten

**Patternlab**
Wordt gebruikt als hoofdnavigatie.
<img width="453" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/53ff8d11-5bea-446c-a86d-41b53708ff2b">

Menu link 
Link die onderdeel is van de Page Header Menu.
<img width="141" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/1921fc28-66e2-4ba2-8713-c526664ea34a">

Menu link active state
<img width="179" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/cbf98c4c-e051-44a2-b929-9abd7e1a1fac">

**Website**
<img width="351" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/5780c54c-9b33-43f2-b942-7debebc7f151">

---

### Notities
- Onder de naam Page Header Menu. 
- Menu links wrappen over de volgende regel.
- Menu links hebben hebben bij default state een blauwe kleur.
- Bij active state hebben menu links een donkergrijze kleur en is een visuele indicator boven het label te zien.
- Op small containers verdwijnt de Navigation bar achter een hamburgermenu. 
- Geen mogelijkheid tot een megamenu.

**Comment by Rozerinay** [comment]
## Rotterdam

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/story/css-utrecht-nav-bar--default)
- [Figma](https://www.figma.com/design/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?node-id=1730-7404&t=ndBDAmk7C1cnDfRc-11)

---

### Screenshots / Varianten

**Storybook**
Default
<img width="549" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/663aeba5-b5fa-47a0-9551-c02853e93f15">

**Figma**
The header navigation can be constructed out of two bars, a top bar and a menu bar. Use the top bar for tertiary links and for a back link if you’re using a product or label of Gemeente Rotterdam. Use the main menubar for the main menu items.
<img width="1314" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/739847a3-8fc5-4d5d-a3cb-124f3eed0898">

---

### Notities
- Onder de naam Nav bar.
- Gaan uit van 'main menu bar' en 'top bar'. 
- Uitwerking in Storybook bestaat alleen uit de menu bar (i.p.v. de top bar).
- Kan bestaan uit een menu item dat kan worden opengeklapt waarna een menu verschijnt.
- Er wordt gebruik gemaakt van een chevron down om het menu te openen.
- Bij active state hebben navigation items een visuele indicator onder het label.
- Op small containers verdwijnt de Navigation bar achter een hamburgermenu.

**Comment by Rozerinay** [comment]
## RVO

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/components/menubar/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-menubar--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/menubar)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=10-26&t=iz50Z8LgKKWJvgQT-11)

---

### Screenshots / Varianten

**Documentatie website**
<img width="604" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/fa578629-6e51-41bf-82e5-988bc5a43187">

**Storybook**
Default
<img width="734" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/01b7e5a3-af6c-49ee-a4f3-a529618a0443">

Active item
<img width="661" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/cef86ad6-a0d8-4783-8322-4abf75d411e2">

Align to right
<img width="715" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/eead7041-1a6a-4268-833b-bdb333ca987c">

Horizontal divider
<img width="721" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/997d5888-68d1-41d7-b5a6-a2475dfff940">

Vertical divider
<img width="721" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/d6eccb0a-9bbf-4a5c-9eb1-9659039a143c">

With icons left
<img width="714" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/0540aa6d-306d-423c-8dc8-0acef06014f2">

With icons right
<img width="716" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/c7f9c4e4-8d83-441e-ba07-133cd44ddba6">

Max width small
<img width="716" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/898ad200-14a4-48be-afd0-633d020db01b">

Max width medium
<img width="717" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/dc7bb626-7b3c-4e4f-ac06-776504923684">

Max width large
<img width="718" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/4683c55b-55e6-45aa-962d-db2e185a98b2">

Size small
<img width="715" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/adce5e1e-bb11-4a5b-af30-faeb359b0495">

Size medium
<img width="714" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/a6651b1f-43c1-4a0e-a934-091dcba76bc4">

Size large
<img width="720" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/ae92b765-52fd-4e78-ba5c-f66a7f80a6f4">

Submenu
Het menu dat opent nadat een menu item is opengeklapt. 
<img width="718" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/de0c9a83-aca6-4d76-8ccf-46bba6efcf82">

Submenu grid
<img width="718" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/4aff3724-dab0-48f7-b5d2-cec3667f9231">


**Figma**
<img width="1110" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/1acd6bc4-61d8-49e5-b201-d8e4f3ceb39e">

**Website**
<img width="921" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/4d3fb6df-30f7-4662-8448-813f307b5ee3">

---

### Notities
- Onder de naam Menu bar.
- Bestaat uit top nav items. 
- Mogelijkheid voor icoon voor of achter link, horizontale of verticale divider, verschillende max widths en voor verschillende sizes. 
- Kan bestaan uit een achtergrondkleur. 
- Uitwerking in de documentatie website, Storybook en Figma verschilt van de website.
- Op de website is er een mogelijkheid om een menu item open te klappen.
- Er wordt gebruik gemaakt van een chevron down om het submenu te openen.
- Op small containers verdwijnt de Navigation bar achter een hamburgermenu.

**Comment by Rozerinay** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-nav-bar--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/nav-bar)

---

### Screenshots / Varianten

**Storybook**

Navigation bar with navigation list
Navigatie die links wordt uitgelijnd.
<img width="655" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/18bc34d8-11b3-4a58-b9fd-2298139f33ee">

Center appearance
Navigatie die in het midden wordt uitgelijnd. 
<img width="1068" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/253bb5c1-ef55-46f6-ab13-6619eac80cb0">

Wrap many items
Veel items die op meerdere regels zijn verspreid.
<img width="1039" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/d3fc0de1-c9de-489c-95ac-b79cbd9b6d77">

Wrap long text in items
Items die bestaan uit lange teksten en daardoor over meerdere regels zijn verspreid. 
<img width="1009" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/82c1ff52-6d80-46de-b3f8-c83c9434d1fc">

**Website**
<img width="1180" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/70c56a21-eb20-4efa-84bf-09f3a208ecc2">

---

## Notities
- Onder de naam Navigation bar.
- Bestaat uit nav list items. 
- Uitwerking in Storybook verschilt van de website.
- Geen mogelijkheid tot een megamenu.
- Geen andere styling voor active state.
- Op small containers verdwijnt op de de Navigation bar achter een hamburgermenu en in Storybook wrapt het over meerdere regels.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Navigation bar op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=navigation+bar)

**Comment by emma-koole** [comment]
## Carbon

[Storybook](https://web-components.carbondesignsystem.com/?path=/story/components-ui-shell--header-base-w-navigation-actions-and-side-nav)

**Storybook**
![Scherm­afbeelding 2025-05-16 om 11 53 10](https://github.com/user-attachments/assets/2a18c48e-dac2-4412-b839-154be7eb3e0e)

## Notities
Carbon Design System gebruikt een overkoepelend UI Shell component. Hierin kun je gebruik maken van de Header Base with navigation, Actions, Side navigation (left panel).

**Comment by RenateRoke** [comment]
## Informatie van kennisbank digitoegankelijk

### Doe dit voor digitale toegankelijkheid

- Kleurcontrast.
- Toetsenbordtoegankelijkheid.
- Focusindicatie en focuszichtbaarheid.
- Koppen en tekstalternatieven.
- Status voor [hulptechnologie.
- Besteed ook aandacht aan de manier waarop het menu opent, sluit en als je uit het menu navigeert. Test dit ook met een screenreader. Lees het artikel [Accessibile mobile navigation](https://www.a11ymatters.com/pattern/mobile-nav/) (Engelstalig) voor een goede start naar een toegankelijk uitklapmenu. Raadpleeg bij twijfel een expert.
